src/sys/socket/mod.rs

```diff
@@ -987,6 +1013,30 @@ impl ControlMessageOwned {
                 let content_type = unsafe { ptr::read_unaligned(p as *const u8) };
                 ControlMessageOwned::TlsGetRecordType(content_type.into())
             },
+            #[cfg(any(linux_android, target_os = "freebsd"))]
+            #[cfg(feature = "net")]
+            (libc::IPPROTO_IP, libc::IP_TTL) => {
+                let ttl: u8 = unsafe { ptr::read_unaligned(p as *const u8) };
+                ControlMessageOwned::Ipv4RecvTtl(ttl)
+            },
+            #[cfg(any(linux_android, target_os = "freebsd"))]
+            #[cfg(feature = "net")]
+            (libc::IPPROTO_IPV6, libc::IPV6_HOPLIMIT) => {
+                let ttl: u8 = unsafe { ptr::read_unaligned(p as *const u8) };
```

Comment:
BTW, do we have any documents on the size that should be used here, I didn't find much about this...

Reply:
I did not find much as well, so I assumed a `u8` would be enough (same as v4) giving us a max of 256 hops.. What do you think?

Reply:
FreeBSD manual has 3 flags documented:

| IP_RECVTTL | IP_RECVTOS | IP_V6_HOPLIMIT|IPV6_TCLASS |
|------------------|--------------------|-------------------------|-------------------|
| u_char         | u_char             | int|             undocumented  |

[RFC 3542 (IPV6)](https://datatracker.ietf.org/doc/html/rfc3542.html) says `IPV6_HOPLIMIT` and `IPV6_TCLASS` should both use `int`:

```
   opt level/    optname/          optval/
      cmsg_level    cmsg_type         cmsg_data[]
      ------------  ------------      ------------------------
      IPPROTO_IPV6  IPV6_PKTINFO      in6_pktinfo structure
      IPPROTO_IPV6  IPV6_HOPLIMIT     int
      IPPROTO_IPV6  IPV6_NEXTHOP      socket address structure
      IPPROTO_IPV6  IPV6_RTHDR        ip6_rthdr structure
      IPPROTO_IPV6  IPV6_HOPOPTS      ip6_hbh structure
      IPPROTO_IPV6  IPV6_DSTOPTS      ip6_dest structure
      IPPROTO_IPV6  IPV6_RTHDRDSTOPTS ip6_dest structure
      IPPROTO_IPV6  IPV6_TCLASS       int
```

For Linux, I am not sure the type to use, maybe we should take a look at the kernel code:<

Reply:
Ok, I have checked the Linux kernel source code:

| IP_TTL | IP_TOS | IP_HOPLIMIT | IP_TCLASS |
|-----------|------------|---------------------|----------------|
| int        |u_char   int | int|int|

ref: 

v4: https://github.com/torvalds/linux/blob/44ef20baed8edcb1799bec1e7ad2debbc93eedd8/net/ipv4/ip_sockglue.c#L63-L72

v6: 

1. https://github.com/search?q=repo%3Atorvalds%2Flinux+path%3A%2F%5Enet%5C%2F%2F+IPV6_HOPLIMIT&type=code  
2. https://github.com/torvalds/linux/blob/44ef20baed8edcb1799bec1e7ad2debbc93eedd8/net/ipv6/ipv6_sockglue.c#L1174

Reply:
I think this is done now.